app/controllers/admin/api/services_controller.rb

```diff
@@ -111,6 +111,7 @@ def destroy
   protected
 
   def can_create
+    authorize!(:create, Service) if current_user
```

Comment:
This way is perfect for me.

Just for the record, in case one thinks that calling `CanCan::Ability#authorize!` and `head :forbidden` in the next line is not cool, a more sophisticated alternative could be:

```ruby
  def can_create
    authorize!(:create, Service) do
      raise CanCan::AccessDenied unless current_account.can_create_service?
    end
  end

  def authorize!(action, subject, *args)
    super if current_user
    yield if block_given?
    subject
  end
```

It won't break other actions and cover well the case for `Admin::Api::ServicesController#create`.

Reply:
I've commented locally the conditional and still all tests of `test/integration/admin/api/services_controller_test.rb` pass :joy: So imo we are lacking a test for the scenario of the account being able to create but not having a current_user.

Reply:
That was what I was afraid of, couldn't we get the user directly from the access_token instead of assuming there's always a `current_user`?
@guicassolato

Reply:
we get the user from the access_token but sometimes we get the account from the api_key instead so there is no current_user. That is the scenario that we are lacking to be tested.